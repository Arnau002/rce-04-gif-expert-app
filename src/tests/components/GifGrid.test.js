import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs'); // Simulació creació component

describe('Test <GifGrid />', () => { 
    
    const category = 'Test';

    test('Es mostra correctament', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('Ha de mostrar items quan es carreguen imatges amb useFetchGif', () => { 
        const gifs = [{
            id:'ABC',
            url: 'https://localhost/img/imgExemple.gif',
            title: 'Qualsevol Títol'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });

        const wrapper = shallow(<GifGrid category={category}/>);
        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
    });

});