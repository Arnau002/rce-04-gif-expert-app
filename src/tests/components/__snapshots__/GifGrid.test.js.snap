// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifGrid /> Es mostra correctament 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test <GifGrid /> Ha de mostrar items quan es carreguen imatges amb useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Test
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Qualsevol TÃ­tol"
      url="https://localhost/img/imgExemple.gif"
    />
  </div>
</Fragment>
`;
